### ISSUE TITLE
Add /query endpoint to pi server

### ISSUE THREAD

**Issue by natduca** [google/trace-viewer#1124]
Hard part is figuring out auth flow.

**Comment by dj2** [comment]
Ask infa how rietveld does it with the oauth cookie?

**Comment by fmeawad** [comment]
Yes, will do

**Comment by anniesullie** [comment]
I think the infra code that runs for rietveld is scattered between:
https://code.google.com/p/chromium/codesearch#chromium/tools/depot_tools/auth.py
https://code.google.com/p/chromium/codesearch#chromium/tools/depot_tools/third_party/upload.py
https://code.google.com/p/chromium/codesearch#chromium/tools/depot_tools/rietveld.py

Other options:
1) For very short term command line access, you could use https://cloud.google.com/appengine/docs/python/tools/remoteapi (restricting it to admin access of course)
2) If you want to make a permanent API, on the server side it's a good idea to consider cloud endpoints: https://cloud.google.com/appengine/docs/python/endpoints (does the server part of oauth2 for you)

**Comment by fmeawad** [comment]
I already have a temp solution (the one I currently use) that uses remoteapi, but it has a set of problems, including having to have a gae sdk installed on the machine and another level of authentication.

I am working towards a solution that uses endpoints, in the long run this will also be used by server scripts as well.

**Comment by natduca** [comment]
Minor heads up: I'd like to get a script out for dogfood of weather report by next friday. That means we'll need this bug fixed a day or two in advance, like Wedesday. Fadi, I know you're swamped, so if you want, feel free to reassign this to me if you want!

**Comment by fmeawad** [comment]
I think you can use the manual script for a dogfood weather report.
If I do not get it done by the end of Today, I will assign it to you.

**Comment by fmeawad** [comment]
Although I made some progress on it, I will go back to the data pipe from tomorrow.

Things I found that needs to be done:
* The oauth2 might not be the way to go as it is mainly designed to grant authorizations to third parties to use your google data. It can still be adapted to our need.
* You will still need a list of approved users that needs to be stored in the datastore. You will also need an interface to admin that list. For every request, get the user info from google servers and check against that list.
* You will need to specify a format for communicating the query between the script and the request handler (I recommend json)

I think the best approach for the authentication problem is to create a page with login required (maybe admin only), 
that page will generate a token,
that needs to be stored on the machine where the script is being run.
The token is passed from the script to the query page, and checked.
(Which is what OAuth2 will do without the unneeded steps, and will allow to check against domain and/or admin list for the user)
